core/src/main/scala/org/bitcoins/core/hd/HDPath.scala

```diff
@@ -1,16 +1,18 @@
 package org.bitcoins.core.hd
 import scala.util.Try
 
-private[bitcoins] trait HDPath[T <: HDPath[T]] extends BIP32Path {
+private[bitcoins] trait HDPath extends BIP32Path {
+
+  protected type NextPath <: HDPath
```

Comment:
Comment on the intention of this?

Reply:
Is there any way we can do something like `this.type` for this? Seems like we just override it immediately in the sub types

Reply:
Added a explanatory comment in `HDPath.scala`